tests/test_ace_fitting.py

```diff
@@ -11,36 +11,13 @@
 import ase.io
 
 from wfl.configset import ConfigSet_in
-from wfl.fit.ace import fit, dict_to_ace_fit_string
-
-have_julia_with_modules = os.system("julia -e 'using ACE, IPFitting'") == 0
-
-def test_dict_to_ace_fit_string():
-    # atomsfile: list of strings
-    # outfile_base: string
-    # solver: list of string and list
-    # weights: dict
-    # cutoff float
-    params = {'atoms_filename': ['dummy.xyz', 'dummy2.xyz'], 'outfile_base': 'dummy', 'solver': [ 'lsqr', [[0.1, 1.0e-6]] ],
-              'weights': {'default' : { 'E' : 1, 'F' : 1, 'V' : 1}}, 'cutoff' : 3.0, 'dry_run': None }
-    s = dict_to_ace_fit_string(params).strip()
-    print('s', s)
-    assert s == '--atoms_filename dummy.xyz dummy2.xyz --outfile_base dummy --solver lsqr \'[[0.1, 1e-06]]\' --weights \'{"default": {"E": 1, "F": 1, "V": 1}}\' --cutoff 3.0 --dry_run'
-
-    # atomsfile: string
-    # outfile_base: string
-    # key: repeated list of strings
-    # weights: dict
-    params = {'atoms_filename': 'dummy.xyz', 'outfile_base': 'dummy',
-              '_repeat_key': True, 'key': [['E', 'REF_energy'], ['F', 'REF_forces']], 'solver': [ 'lsqr', [[0.1, 1.0e-6]] ],
-              'weights': {'default' : { 'E' : 1, 'F' : 1, 'V' : 1}} }
-    s = dict_to_ace_fit_string(params).strip()
-    print('s', s)
-    assert s == '--atoms_filename dummy.xyz --outfile_base dummy --key E REF_energy --key F REF_forces --solver lsqr \'[[0.1, 1e-06]]\' --weights \'{"default": {"E": 1, "F": 1, "V": 1}}\''
+from wfl.fit.ace import fit, prepare_params_and_fit
 
+have_julia_with_modules = os.system("julia -e  'using ACE1pack'") == 0
```

Comment:
I was just thinking that the test can get done inside the testing function. That way if the testing function is run remotely it'll just fail at that point. I'm not sure there needs to be a separate test for the availability of julia before the function that tests the actual fit call.

Reply:
I see. I have updated the code, but I'm yet to test if the remote pytest.skip behaves correctly (likely not...).

Reply:
If I'm right that pytest.skip uses exceptions, and if you can pickle exceptions, we can make it work.  I hadn't thought about it, but if instead of printing the exception we pickle it, and then the local function creates it by unpickling and raising the resulting object, it might just work.

Reply:
Let's not worry about this now, and hope the pytest.skipif will work. I'll play with the exceptions in expyre later.